eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/io/JsonNodeToClusteredVariantProcessor.java

```diff
@@ -32,74 +31,70 @@
 import java.time.format.DateTimeFormatter;
 import java.util.function.Function;
 
+/**
+ * Processes a DbSNP JsonNode to produce a clustered variant entity.
+ */
 public class JsonNodeToClusteredVariantProcessor implements ItemProcessor<JsonNode, DbsnpClusteredVariantEntity> {
 
-    private static String DEFAULT_VALUE_STRING = "-";
     private static Logger logger = LoggerFactory.getLogger(JsonNodeToClusteredVariantProcessor.class);
     private Function<IClusteredVariant, String> hashingFunction =
-            new ClusteredVariantSummaryFunction().andThen(new SHA1HashingFunction());
+        new ClusteredVariantSummaryFunction().andThen(new SHA1HashingFunction());
 
     public JsonNodeToClusteredVariantProcessor() {
     }
 
-    public DbsnpClusteredVariantEntity process(JsonNode jsonNode) {
-        return parseJsonNodeToDbsnpClusteredVariantEntity(jsonNode);
-    }
-
     /**
      * Parses the JSON into a DbsnpClusteredVariantEntity
      * @param jsonRootNode root node of a dbSNP JSON line item
      * @return dbsnpClusteredVariantEntity
      */
-    private DbsnpClusteredVariantEntity parseJsonNodeToDbsnpClusteredVariantEntity(JsonNode jsonRootNode) {
+    public DbsnpClusteredVariantEntity process(JsonNode jsonRootNode) {
         long accession = jsonRootNode.path("refsnp_id").asLong();
         ClusteredVariant clusteredVariant = parseJsonNodeToClusteredVariant(jsonRootNode);
         String hashedMessage = hashingFunction.apply(clusteredVariant);
         return new DbsnpClusteredVariantEntity(accession, hashedMessage, clusteredVariant);
     }
 
     private ClusteredVariant parseJsonNodeToClusteredVariant(JsonNode jsonRootNode) {
-        String assemblyAccession, contig;
-        assemblyAccession = contig = DEFAULT_VALUE_STRING;
-        // @see https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=9606
+        // @see <a href=https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=9606>Human Tax ID.</a>
         int taxonomyAccession = 9606;
-        long start = 0L;
 
         // JSON date in ISO-8601 format
         DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-d'T'HH:mm'Z'");
         LocalDateTime createdDate = LocalDateTime.parse(jsonRootNode.path("create_date").asText(), formatter);
+
         VariantType type = prepareVariantType(jsonRootNode.path("primary_snapshot_data").path("variant_type").asText());
+
         JsonNode infoNode = jsonRootNode.path("primary_snapshot_data").path("placements_with_allele");
         for(JsonNode alleleInfo : infoNode) {
+
             boolean isPtlp = alleleInfo.path("is_ptlp").asBoolean();
             JsonNode assemblyInfo = alleleInfo.path("placement_annot").path("seq_id_traits_by_assembly");
+            JsonNode spdi = alleleInfo.path("alleles").get(0).path("allele").path("spdi");
```

Comment:
This variant isn't used before the immediately following conditions, so please declare it at the first point of usage.